### PR TITLE
update dependencies

### LINKED ISSUE #45: Dependency '@react-native-community/async-storage' is deprecated
As stated [here](https://www.npmjs.com/package/@react-native-community/async-storage), '@react-native-community/async-storage' moved to a new organization, therefore the dependency must be updated in package.json

### PR DESCRIPTION
**Pull request by mbastian93**
* update some dependencies
* remove @react-native-community/async-storage
* add react-native-encrypted storage (fixes #45 )
*  configure mock

### CODE REVIEW

**Comment by GolexPrime** [comment]
First round of `npm clean_install_macos` brought me this (imagine it printed in red onto you console):
```
[!] CocoaPods could not find compatible versions for pod "Firebase/Messaging":
  In Podfile:
    RNFBMessaging (from `../node_modules/@react-native-firebase/messaging`) was resolved to 12.7.5, which depends on
      Firebase/Messaging (= 8.6.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Messaging (= 8.6.0)`.
```
`pod install --repo-update`   seems to resolve that one, but I do not have finished testing the app after that command yet (but looking good so far).

I would only recommend updating some comments and the README to mirror the change.

File: __mocks__/mocks.js [hunk 1]
```diff
@@ -25,7 +24,7 @@ jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
 
 // mocks the async storage
 /*-----------------------------------------------------------------------------------*/
-jest.mock("@react-native-community/async-storage", () => mockAsyncStorage);
+// jest.mock("@react-native-community/async-storage", () => mockAsyncStorage);
```
Comment (GolexPrime) [inline __mocks__/mocks.js:27]: Could we remove this?

File: src/config/appConfig.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ const conf = {
 
   /** dev-option:
    * skips the QR login after 500ms */
-  automateQrLogin: __DEV__ && false,
```
Comment (GolexPrime) [inline src/config/appConfig.js:66]: I wouldn't mind leaving that one as it is. But the default was `false` so far.
Was that on purpose?

**Review by GolexPrime — COMMENTED** [review]
- Looking good and working properly. 

- I had an issue with updating my dependencies but an additional command cleared that one out. Aside from that it's running well - might be worth mentioning somewhere. 

- There are still some remnants of AsyncSctorage to find in the code that should be removed in future commits, like:
  - `docs/appConfiguration/README.adoc`
  - `android/app/app.iml`
  - `checkInReducer.js`
  - `appConfig.js`
  - `loginActions.js`

File: src/config/appConfig.js (hunk 2 above)
Comment (mbastian93) [reply src/config/appConfig.js:66]: No, it wasn't😅

File: __mocks__/mocks.js (hunk 1 above)
Comment (mbastian93) [reply __mocks__/mocks.js:27]: Oh yeah, sure

**Review by mahvaezi — APPROVED** [review]
Please Apply your changes to Readme.

**Review by mahvaezi — APPROVED** [review]

**Pull request merged**